### PR TITLE
feat: embedded db

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/config-db/issues/305

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+{{- end }}
 metadata:
   name: {{ include "config-db.name" . }}
   labels: {{- include "config-db.labels" . | nindent 4 }}
 spec:
   replicas: {{ .Values.replicas }}
   selector:
     matchLabels: {{- include "config-db.selectorLabels" . | nindent 6 }}
+  {{- if and (eq .Values.db.external.enabled false) (eq .Values.db.embedded.persist true) }}
+  serviceName: {{ include "config-db.name" . }}
```
Comment (moshloop) [inline chart/templates/deployment.yaml:15]:
```suggestion
  serviceName: {{ include "config-db.name" . }}
  {{- if and (eq .Values.db.external.enabled false) (eq .Values.db.embedded.persist true) }}
```

File: chart/values.yaml [hunk 2]
```diff
@@ -4,34 +4,43 @@
...
 
 configChangeRetentionDays: 60
 configAnalysisRetentionDays: 60
 
 # a list of configmaps to load scrape rules from, the configmap should have a single entry called "config.yaml"
 scrapeRuleConfigMaps:
   - config-db-rules
+
 db:
-  # Setting this to true will create a postgres stateful set for config-db to connect to.
```
Comment (moshloop) [inline chart/values.yaml:25]: We seem to have lost the ability to deploy a postgres statefulset ?

Comment (adityathebe) [reply chart/values.yaml:25]: I think you can still set `db.external.enabled: true` ?

Comment (adityathebe) [reply chart/values.yaml:25]: need to add `create`

**Comment by adityathebe** [comment]
@moshloop I'm not really sure why we get `operation not permitted` error when using embedded database.

- The Dockerfile correctly has the owner of `/opt/database` directory set to the new `catalog(1000)` user. I tested this on a docker container and the permission is correctly set.
- Inside Kubernetes, since we're mounting a volume to `/opt/database`, I have also set `fsGroup` to `1000` in the security context. Can also confirm that the process is run as `catalog` user.

Despite this, the error still occurs. 
Do you see anything obvious that I'm missing?

**Comment by adityathebe** [comment]
@moshloop This is ready to go

**Pull request merged**